saleor/webhook/payloads.py

```diff
@@ -301,43 +309,55 @@ def generate_product_deleted_payload(product: "Product", variants_id):
 
 
 PRODUCT_VARIANT_FIELDS = (
-    "name",
     "sku",
+    "name",
+    "track_inventory",
     "private_metadata",
     "metadata",
 )
 
 
-def generate_product_variant_payload(product_variant: "ProductVariant"):
+def generate_product_variant_listings_payload(variant_channel_listings):
+    serializer = PayloadSerializer()
+    fields = (
+        "currency",
+        "price_amount",
+        "cost_price_amount",
+    )
+    channel_listing_payload = serializer.serialize(
+        variant_channel_listings,
+        fields=fields,
+        extra_dict_data={"channel_slug": lambda vch: vch.channel.slug},
+    )
+    return channel_listing_payload
+
+
+def generate_product_variant_media_payload(product_variant):
+    return [
+        {
+            "alt": media_obj.media.alt,
+            "url": (
+                build_absolute_uri(media_obj.media.image.url)
+                if media_obj.media.type == ProductMediaTypes.IMAGE
+                else media_obj.media.external_url
+            ),
+        }
+        for media_obj in product_variant.variant_media.all()
+    ]
+
+
+def generate_product_variant_payload(product_variants: Iterable["ProductVariant"]):
     serializer = PayloadSerializer()
-    product_id = graphene.Node.to_global_id("Product", product_variant.product.id)
     payload = serializer.serialize(
-        [product_variant],
+        product_variants,
         fields=PRODUCT_VARIANT_FIELDS,
-        additional_fields={
-            "channel_listings": (
-                lambda p: p.channel_listings.all(),
-                (
-                    "currency",
-                    "price_amount",
-                    "cost_price_amount",
-                ),
-            ),
-        },
         extra_dict_data={
-            "attributes": serialize_product_or_variant_attributes(product_variant),
-            "product_id": product_id,
-            "media": [
-                {
-                    "alt": media_obj.media.alt,
-                    "url": (
-                        build_absolute_uri(media_obj.media.image.url)
-                        if media_obj.media.type == ProductMediaTypes.IMAGE
-                        else media_obj.media.external_url
-                    ),
-                }
-                for media_obj in product_variant.variant_media.all()
-            ],
+            "attributes": lambda v: serialize_product_or_variant_attributes(v),
+            "product_id": lambda v: graphene.Node.to_global_id("Product", v.product_id),
+            "media": lambda v: generate_product_variant_media_payload(v),
+            "channel_listings": lambda v: json.loads(
+                generate_product_variant_listings_payload(v.channel_listings.all())
```

Comment:
I think we should prefetch `channel` here.

Reply:
We cannot add prefetch for deleted objects. I created ticket for optimalization.